### PR TITLE
FIX: fix stop_session func bug

### PR DESCRIPTION
**Pull request by yunzhongyan0**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/serve/async_engine.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def stop_session(self, session_id: int):
                 input_ids,
                 request_output_len=0,
                 sequence_start=False,
-                sequence_end=False,
+                sequence_end=True,
```
Comment (lvhan028) [inline lmdeploy/serve/async_engine.py:58]: This is not a bug.
`stop_session` is used to stop generation, but not to destroy its history conversations that are cached in server.
but `sequence_end=True` is used to terminate the session, including its conversations. 
PR #544 adds a new API `end_session` to deal with the issue

**Review by AllentDan — APPROVED** [review]
LGTM

**Comment by AllentDan** [comment]
Shall we rename the behavior `stop` to `pause` and `end` to `destroy`? This is clear to me.

**Review by lvhan028 — APPROVED** [review]

**Comment by lvhan028** [comment]
> Shall we rename the behavior `stop` to `pause` and `end` to `destroy`? This is clear to me.
`stop` is to abort the session that is generating tokens. When a session is stopped, he cursor `step` should be rollback. At this point, it is not `pause`.
`end` comes from triton inference server.
I think they make some sense.

**Pull request merged**